tests/pylint/test_imports.py

```diff
@@ -121,3 +121,40 @@ def test_bad_import(
         ),
     ):
         imports_checker.visit_importfrom(import_node)
+
+
+def test_bad_root_import(linter: UnittestLinter, imports_checker: BaseChecker) -> None:
+    """Ensure bad root imports are rejected."""
+
+    first_node, second_node = astroid.extract_node(
+        """
+    import homeassistant.components.climate.const as climate #@
+    from homeassistant.components.climate.const import ClimateEntityFeature #@
```

Comment:
A few good examples. Feel free to add more
```py
from homeassistant.components import climate
from homeassistant.components.climate import ClimateEntityFeature
```

A bad example which should catch the `"const" in node.names` error.
```py
from homeassistant.components.climate import const
```

Reply:
Done - split into two separate test functions